pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -429,12 +434,18 @@ func (vm *volumeManager) WaitForAttachAndMount(ctx context.Context, pod *v1.Pod)
 			return nil
 		}
 
-		return fmt.Errorf(
+		err = fmt.Errorf(
 			"unmounted volumes=%v, unattached volumes=%v, failed to process volumes=%v: %w",
 			unmountedVolumes,
 			unattachedVolumes,
 			volumesNotInDSW,
 			err)
+
+		// Record an event for the pod, usually events are sent because some volumes of the pod are not mounted successfully.
+		// If the volume is not mounted successfully for a long time, events need to be sent to remind the user what happened, see #127004
+		vm.recorder.Eventf(pod, v1.EventTypeWarning, events.FailedMountVolume, "Failed to mount pod volumes: %v", err)
```

Comment:
The message is not entirely correct. It's not just about mounting, but also about attaching volumes.

Reply:
https://github.com/kubernetes/kubernetes/blob/f836773540cc83488250a12637e7d924078ef284/pkg/kubelet/volumemanager/volume_manager.go#L426-L431

hi @bart0sh , I think the volumes that failed to attach can be in the err message, and the message only needs to tell `mounting`. regardless of whether there are unattached volumes, as long as unmount volumes == 0, it is considered to be no error.